flow/project.py

```diff
@@ -1455,11 +1466,13 @@ def __call__(self, func, name=None):
                 # delay setting the aggregator because we do not restrict the decorator
                 # placement in terms of `@FlowGroupEntry`, `@aggregator`, or
                 # `@operation`.
-                self._parent_class._GROUPS.append(FlowGroupEntry(name=name, options=""))
+                self._parent_class._GROUPS.append(
+                    FlowGroupEntry(name=name, project=self._parent_class, options="")
+                )
                 if hasattr(func, "_flow_groups"):
-                    func._flow_groups.append(name)
+                    func._flow_groups.add(name)
                 else:
-                    func._flow_groups = [name]
+                    func._flow_groups = {name}
```

Comment:
It should always be the case now that `_flow_groups` does not exist. If it does exist, it should be an error, unless we are allowing different parent classes to also declare an operation without error.

Reply:
We should still keep the above check though because another project could set the `_flow_groups` attribute.

Reply:
@b-butler I'm confused, is there an action item here? I think after your second comment that the answer is no.

Reply:
I think we should error if `func._flow_groups` exists already.